csharp/test/Apache.Arrow.Adbc.Tests/ClientTests.cs

```diff
@@ -304,5 +331,49 @@ static void AssertTypeAndValue(
                 Assert.True(ctv.ExpectedValue == null, Utils.FormatMessage($"The value for {ctv.Name} is null and but it's expected value is not null for query [{query}]", environmentName));
             }
         }
+
+        static bool AreExpandoObjectsEqual(ExpandoObject? obj1, ExpandoObject? obj2)
+        {
+            if (obj1 == null && obj2 == null)
+            {
+                return true;
+            }
+            else if (obj1 != null && obj2 == null)
+            {
+                return false;
+            }
+            else if (obj1 == null && obj2 != null)
+            {
+                return false;
+            }
+
+            var dict1 = (IDictionary<string, object?>)obj1!;
+            var dict2 = (IDictionary<string, object?>)obj2!;
+
+            if (dict1.Count != dict2.Count)
+                return false;
+
+            foreach (var key in dict1.Keys)
```

Comment:
I know this is test code and therefore non-performance-critical, but I also think it's good to default to performant code when there's no cost to readability. (Plus, I feel like I've written this function dozens of times and so this stands out for me :).)

This can be done more efficiently with
```
foreach (var pair in dict1)
{
    if (!dict2.TryGetValue(pair.Key, out object? value2))
        return false;

    ...
}
```

Reply:
fixed in latest push